### PR TITLE
WIP: allow for multiple cutoffs (nx.all_simple_paths)

### PR DESCRIPTION
**Pull request by harisbal**
Allow limiting the results of nx.all_simple_paths not only by the length of the path but as well as by other weight attributes.

### CODE REVIEW

**Comment by harisbal** [comment]
Any idea why I received this [error](https://travis-ci.org/networkx/networkx/jobs/600370976) ?
Cheers

**Comment by jarrodmillman** [comment]
Travis hung for some reason unrelated to your changes.  I restarted the test and it worked fine this time.

File: networkx/algorithms/simple_paths.py [hunk 1]
```diff
@@ -106,14 +106,18 @@ def all_simple_paths(G, source, target, cutoff=None):
...
+       If a dictionary is passed then different cutoffs can be used for different weights.
+       (e.g. {'weight1': cutoff_value1, 'weight2': cutoff_value2})
+       The depth of search can be specified by using `None` as key (e.g. {None: cutoff})
 
     Returns
     -------
     path_generator: generator
        A generator that produces lists of simple paths.  If there are no paths
-       between the source and target within the given cutoff the generator
+       between the source and target within the given cutoff_len the generator
```
Comment (jarrodmillman) [inline networkx/algorithms/simple_paths.py]: Did you mean to change this?

File: networkx/algorithms/tests/test_simple_paths.py [hunk 2]
```diff
@@ -112,6 +112,57 @@ def test_digraph_all_simple_paths_with_two_targets_cutoff():
...
+def test_all_simple_paths_weighted_graph_with_weight_cutoff():
+    edges = [(1, 2), (2, 10), (1, 5), (5, 4), (4, 3), (3, 10)]
+    distances = [5, 6, 1, 3, 2, 2]
+    d = {e: {'Distance': dist} for e, dist in zip(edges, distances)}
+
+    G = nx.Graph(edges)
+    nx.set_edge_attributes(G, d)
+
+    paths = list(nx.all_simple_paths(G, 1, 10, cutoff={'Distance': 10}))
+    assert len(paths), 1
```
Comment (jarrodmillman) [inline networkx/algorithms/tests/test_simple_paths.py]: Do you want to check if `len(paths)` is `1`?  If so, this should be `assert len(paths) == 1`.  You may want to review this:
http://doc.pytest.org/en/latest/assert.html

File: networkx/algorithms/simple_paths.py (hunk 1 above)
Comment (harisbal) [reply networkx/algorithms/simple_paths.py]: nope, it's just a remnant from older version

File: networkx/algorithms/tests/test_simple_paths.py (hunk 2 above)
Comment (harisbal) [reply networkx/algorithms/tests/test_simple_paths.py]: I think that was the nose way to assert stuff. Thanks for the link

**Comment by harisbal** [comment]
Hello,
Any idea why this failed. The failing build on AppVeyor did not even start.
> Cannot fail over job to clouds/groups 'pro-backup'. All clouds within the group are either failing, offline, busy or reached build minutes quota.

Do you know if I should push again to force a build?
Cheers

**Comment by MridulS** [comment]
Looks like git has messed up things here 😅 

Feel free to send in a new PR with the changes if you still have the interest and bandwidth (no worries if you dont!) :)
Thanks again for your contributions @harisbal!

**Pull request closed without merging**